rust/worker/src/server.rs

```diff
@@ -263,8 +265,41 @@ impl chroma_proto::metadata_reader_server::MetadataReader for WorkerServer {
             }
         };
         println!("Querying count for segment {}", segment_uuid);
-        // TODO: Add logic here to count.
-        let response = CountRecordsResponse { count: 0 };
+        let dispatcher = match self.dispatcher {
+            Some(ref dispatcher) => dispatcher,
+            None => {
+                return Err(Status::internal("No dispatcher found"));
+            }
+        };
+
+        let system = match self.system {
+            Some(ref system) => system,
+            None => {
+                return Err(Status::internal("No system found"));
+            }
+        };
+
+        let orchestrator = CountQueryOrchestrator::new(
+            system.clone(),
+            &segment_uuid,
+            self.log.clone(),
+            self.sysdb.clone(),
+            dispatcher.clone(),
+            self.blockfile_provider.clone(),
+        );
+
+        let result = orchestrator.run().await;
+        let c = match result {
+            Ok(r) => {
+                println!("(Sanket-temp) Count value {}", r);
```

Comment:
I think we should print the value here but maybe we can remove `(Sanket-temp)` :)

Reply:
oops, will do. Thank you!